src/utils/PushQueue.js

```diff
@@ -259,14 +279,17 @@ export default class PushQueue {
                     continue // eslint-disable-line no-continue
                 }
 
-                const value = await new Promise((resolve, reject) => {
+                const deferred = new Promise((resolve, reject) => {
                     // wait for next push
                     this.nextQueue.push({
                         resolve,
                         reject,
                     })
                 })
 
+                deferred.catch(() => {}) // prevent unhandledrejection
```

Comment:
Should  the `catch` be chained to the the `deferred` object? If we just call it, it creates a new promise and doesn't change the original.

Reply:
This is intentional. Adding a `.catch` here prevents `node --unhandled-rejections=strict` (enabled by default in next major node version) from crashing with an `unhandledrejection` when outstanding queued promises are cancelled. i.e. these particular promises are allowed to reject quietly and be handled asynchronously.